WIP: Re-organize and add tests

jasonpincin opened the pull request:
This PR Re-organizes the repo to bring it in line with the Autopilot pattern direction and includes tests using [autopilotpattern/testing](https://github.com/autopilotpattern/testing).

tgross commented on test/tests.py:
```diff
@@ -0,0 +1,117 @@
+        replica_containers = self.get_service_instances_from_consul('redis-replica')
+        value = uuid.uuid4()
+        self.docker_exec(master_container, 'redis-cli set test:repl ' + str(value))
+        for replica in replica_containers:
```
This replication is async, isn't it? It's very unlikely that we'll lose this race but we might want to try to account for it. Is there a way to verify that the replica is up-to-date?

jasonpincin replied:
It is and yeah I thought about that but didn't account for it on the first pass. Simplest way would probably be a `wait_for_replication` function that polled the replica for the expected value up to a timeout, as we've done other places.

tgross commented:
@jasonpincin we should namespace the CNS entry for Consul to `redis-consul`, so as not to conflict with other Consul instances that might be running in the same Triton account.

jasonpincin commented:
@tgross I addressed both your concerns with this latest round of updates. Implemented a `wait for` on the replication test and name-spaced `consul`. I'm still working on getting the timeouts correct for Jenkins, so it's not quite ready yet, but don't expect any large additional changes if you want to give it another pass. I would like to discuss the pattern I'm using here for local vs cloud testing.

tgross commented on README.md:
```diff
@@ -2,9 +2,9 @@
-[![ImageLayers](https://badge.imagelayers.io/faithlife/redis:autopilot.svg)](https://imagelayers.io/?images=faithlife/redis:autopilot)
+[![DockerPulls](https://img.shields.io/docker/pulls/autopilotpattern/redis.svg)](https://registry.hub.docker.com/u/autopilotpattern/redis/)
+[![DockerStars](https://img.shields.io/docker/stars/autopilotpattern/redis.svg)](https://registry.hub.docker.com/u/autopilotpattern/redis/)
+[![ImageLayers](https://badge.imagelayers.io/autopilotpattern/redis.svg)](https://imagelayers.io/?images=autopilotpattern/redis)
```
Just FYI we've found this service to be a little flaky so we've cut this badge from most of the other repos.

tgross commented on README.md:
```diff
@@ -2,9 +2,9 @@
+[![DockerPulls](https://img.shields.io/docker/pulls/autopilotpattern/redis.svg)](https://registry.hub.docker.com/u/autopilotpattern/redis/)
+[![DockerStars](https://img.shields.io/docker/stars/autopilotpattern/redis.svg)](https://registry.hub.docker.com/u/autopilotpattern/redis/)
+[![ImageLayers](https://badge.imagelayers.io/autopilotpattern/redis.svg)](https://imagelayers.io/?images=autopilotpattern/redis)
 [![Join the chat at https://gitter.im/autopilotpattern/general](https://badges.gitter.im/autopilotpattern/general.svg)](https://gitter.im/autopilotpattern/general)
```
I don't think we're maintaining a presence here anymore, really.

tgross commented on Makefile:
```diff
@@ -0,0 +1,182 @@
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		-e TAG=$(tag) \
+		-e COMPOSE_FILE=compose/docker-compose.yml \
+		-e PATH=/root/venv/3.5/bin:/bin:/usr/bin:/usr/local/bin \
```
I think you can drop this -- we're not installing Python3 into a virtualenv.

tgross commented:
LGTM!
